code/modules/paperwork/ticketmachine.dm

```diff
@@ -117,52 +131,60 @@
 			maptext_x = 13
 		if(10 to 99)
 			maptext_x = 10
-		if(100 to 999)
+		if(100)
 			maptext_x = 8
 	maptext = "[current_number]" //Finally, apply the maptext
 
 /obj/machinery/ticket_machine/attackby(obj/item/I, mob/user, params)
 	..()
-	if(ticket_number >= max_number)
-		to_chat(user, "<span class='notice'>[src] refuses [I]!, perhaps it's already full?.</span>")
-		return
 	if(istype(I, /obj/item/hand_labeler_refill))
+		if(!(ticket_number >= max_number))
+			to_chat(user, "<span class='notice'>[src] refuses [I]! There [max_number-ticket_number==1 ? "is" : "are"] still [max_number-ticket_number] ticket\s left!</span>")
+			return
 		to_chat(user, "<span class='notice'>You start to refill [src]'s ticket holder (doing this will reset its ticket count!).</span>")
 		if(do_after(user, 30, target = src))
-			to_chat(user, "<span class='notice'>You insert [I] into [src] as it whirrs nondescriptly.</span>")
+			to_chat(user, "<span class='notice'>You insert [I] into [src] as it whirs nondescriptly.</span>")
 			qdel(I)
 			ticket_number = 0
 			current_number = 0
+			if(tickets.len)
+				for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+					ticket.visible_message("<span class='notice'>\the [ticket] disperses!</span>")
+					qdel(ticket)
+				tickets.Cut()
 			max_number = initial(max_number)
 			update_icon()
 			return
-	if(istype(I, /obj/item/ticket_machine_ticket))
-		to_chat(user, "<span class='warning'>You start to cram [I] into [src]'s recycling bin.</span>")
-		if(do_after(user, 20, target = src)) //Slight delay so they don't accidentally dispose of their ticket and move to the BACK OF THE LINE
-			qdel(I)
-			return
 
 /obj/machinery/ticket_machine/proc/reset_cooldown()
 	ready = TRUE
 
 /obj/machinery/ticket_machine/attack_hand(mob/living/carbon/user)
 	. = ..()
-	if(!ready)
-		to_chat(user,"Temporarily unable to dispense ticket, please be patient!")
+	if(!ready && (obj_flags & EMAGGED))
```

Comment:
is this suppose to be OR

Reply:
or `!(obj_flags & EMAGGED)`?

Reply:
Only need the cooldown when it's emagged because when it's emagged you can print unlimited tickets (normally it's only one per person), and in testing that leads to huge piles of burning tickets which I thought would be laggy and undesirable.

Reply:
You already have the cooldown incorporated for all tickets, may as well remove the emagged check.

Reply:
Say there's three people each getting a ticket. Without the cooldown the second person doesn't have to wait after the first and so on. And it can't be abused because until the ticket is destroyed they can't get another one. When it's emagged you want the cooldown to prevent people from printing flaming tickets as fast as they can click (they drop to the ground automatically).

Reply:
ok then toss the cooldown stuff under emagging and remove the emag check above?

Reply:
@ExcessiveUseOfCobblestone done